app/services/api_get_item_metadata.rb

```
@@ -14,19 +14,31 @@ def get_data!
     validate_input!
 
     params = "barcode=#{@barcode}"
-    raw_results = ApiHandler.call("GET", @path, nil, params)
-    results = {"status" => raw_results["status"], "results" => 
-      {
-        "title" => raw_results["results"]["title"],
-        "author" => raw_results["results"]["author"],
-        "chron" => raw_results["results"]["description"],
-        "bib_number" => raw_results["results"]["bib_id"],
-        "isbn_issn" => raw_results["results"]["isbn_issn"],
-        "conditions" => raw_results["results"]["condition"],
-        "call_number" => raw_results["results"]["call_number"]
-      }}
-
-     results
+    begin
+      raw_results = ApiHandler.call("GET", @path, nil, params)
+      results = {"status" => raw_results["status"], "results" => 
+        {
+          "title" => raw_results["results"]["title"],
+          "author" => raw_results["results"]["author"],
+          "chron" => raw_results["results"]["description"],
+          "bib_number" => raw_results["results"]["bib_id"],
+          "isbn_issn" => raw_results["results"]["isbn_issn"],
+          "conditions" => raw_results["results"]["condition"],
+          "call_number" => raw_results["results"]["call_number"]
+        }}
+    rescue Timeout::Error => e  # Status code 599 is a MS defined code for network timeout
```

Timeout is a predictable potential error, so worth trapping and coding for. I don't know what other errors might happen here, so swallowing them up probably isn't desirable.